### DISCUSSION TITLE
How to activate SSH / enable SSH port

### DISCUSSION THREAD

**Discussion by traptoo** [tteck/Proxmox#764] [Help Needed]
I tried to install the SSH addon with nearly every tutorial found in the web. Non of them seemed to work with your install script.

The Log shows:
`WARNING: SSH port is disabled. Prevent start of SSH server.`

Could you explain, how to get SSH to work with your proxmox install script?

Environment:
Proxmox 7.2

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/discussions/385

**Comment by LostPoE** [comment]
You should be able to create ~/.ssh/authorized_keys and put your .pub key in there. Then you don't have you use a password for login.

**Comment by traptoo** [comment]
Thanks for your reply.
Your solution is not totally clear to me.
Where do i have to create the ~/.ssh/authorized_keys? In Proxmox shell? In the lxc console? in HA file editor?
Sorry for my basic questions, but lxc is quite new to me.

Eventually it could help if i would descripe my goal i want to achieve:
I need a unofficial addon to support one device in HA. Therefore it is neccessary to get the HACS addon store (to install the needed addon) which needs ssh to install.
HA was installed by this script:
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/vm/haos-vm-v4.sh)"`

Actual situation is that i cannot start the ssh addon (it stopped immidiatly). The log shows following:

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/apks.sh
cont-init: info: /etc/cont-init.d/apks.sh exited 0
cont-init: info: running /etc/cont-init.d/keygen.sh
[11:30:12] INFO: Generating host keys...
ssh-keygen: generating new host keys: RSA DSA ECDSA ED25519 
cont-init: info: /etc/cont-init.d/keygen.sh exited 0
cont-init: info: running /etc/cont-init.d/profile.sh
cont-init: info: /etc/cont-init.d/profile.sh exited 0
cont-init: info: running /etc/cont-init.d/ssh.sh
cont-init: info: /etc/cont-init.d/ssh.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun sshd (no readiness notification)
services-up: info: copying legacy longrun ttyd (no readiness notification)
**[11:30:12] WARNING: SSH port is disabled. Prevent start of SSH server.**
s6-rc: info: service legacy-services successfully started
[11:30:12] INFO: Starting Web Terminal...
[2022/11/23 11:30:12:5339] N: ttyd 1.6.3-3e37e33 (libwebsockets 4.2.1-v4.2.1)
[2022/11/23 11:30:12:5340] N: tty configuration:
[2022/11/23 11:30:12:5340] N:   start command: tmux -u new -A -s homeassistant bash -l
[2022/11/23 11:30:12:5340] N:   close signal: SIGHUP (1)
[2022/11/23 11:30:12:5340] N:   terminal type: xterm-256color
[2022/11/23 11:30:12:5340] N: LWS: 4.2.1-v4.2.1, loglevel 7
[2022/11/23 11:30:12:5340] N: NET CLI SRV H1 H2 WS ConMon IPV6-off
[2022/11/23 11:30:12:5350] N:    /usr/lib/libwebsockets-evlib_uv.so
[2022/11/23 11:30:12:5407] N:  Using foreign event loop...
[2022/11/23 11:30:12:5407] N:  ++ [wsi|0|pipe] (1)
[2022/11/23 11:30:12:5408] N:  ++ [vh|0|netlink] (1)
[2022/11/23 11:30:12:5408] N:  ++ [vh|1|default||8099] (2)
[2022/11/23 11:30:12:5409] N: lws_socket_bind: nowsi: source ads 0.0.0.0
[2022/11/23 11:30:12:5409] N:  ++ [wsi|1|listen|default||8099] (2)
[2022/11/23 11:30:12:5414] N:  Listening on port: 8099

**Comment by tteck** [comment]
> I need a unofficial addon to support one device in HA. Therefore it is neccessary to get the HACS addon store (to install the needed addon) which needs ssh to install.
HA was installed by this script:
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/vm/haos-vm-v4.sh)"`

You need to install one of the SSH add-ons in the Home Assistant [Add-on store](https://my.home-assistant.io/redirect/supervisor_store/) (you need to enable advanced mode in your user profile to see them)
https://hacs.xyz/docs/setup/download/

**Reply by traptoo** [reply]
Got the solution!

Shame on me:
The only thing to do was to reboot proxmox. After that the SSH addon started automatically and the configuration tab get active.
